app/src/main/java/com/google/droidjump/models/GameItem.java

```diff
@@ -14,25 +14,26 @@
  * limitations under the License.
  */
 
-package com.google.droidjump;
+package com.google.droidjump.models;
 
+import android.content.res.Resources;
 import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
 
-public class DrawableElement {
-
+abstract class GameItem {
     protected int x;
     protected int y;
-    protected Bitmap bitmap;
+    protected Bitmap picture;
 
-    public DrawableElement(int x, int y, Bitmap bitmap) {
+    public GameItem(int x, int y, int pictureID, Resources resources) {
         this.x = x;
         this.y = y;
-        this.bitmap = bitmap;
+        picture = BitmapFactory.decodeResource(resources, pictureID);
+        correctYCoordinate();
```

Comment:
I think that this class should not change coordinates by itself. It might be very confusing to create a GameItem with "y=100" and get "item.getY()" from it right after just to discover that it changed for some reason.

I would suggest you to add method to this class like "getYWithBitmapOffset" which would return "y - picture.getHeight()" value. Should be clearer this way.

Reply:
I understand the problem. But is it ok to never use getY() method? And there could be a confusion for implementing other logic. For example collision checks. Because to check the intersection obstacle and Droid we need to know the coordinates and in all cases when we want to know coordinates we have to use getYWithBitmapOffset() because it's just correct location on a screen and we don't need incorrect getY().

Reply:
It might be OK.

As an alternative, if you only need corrected coordinate you can instead just create objects with the corrected "y" coordinate.

Reply:
Yes. I was thinking about it, but I don't know, how to do it better. To set the already corrected coordinate to the constructor, we need to know the bitmap height outside of class. And it is not cool to create special Bitmap for that. It is possible to do like this:
Droid droid = new Droid(x, y - Droid.height);
if height is a static field, but there is a problem with initialisation (don't know how to set value to static final field from resources, and also I should change logic of super classes)

Could you suggest something? Or I just can implement option with getYWithBitmapOffset() method

Reply:
You can implement "getYWithBitmapOffset()" - probably not ideal but works. Another suggestion is to change constructor to:
```
public GameItem(int x, int yWithBitmapOffset, int pictureID, Resources resources) {
```

Maybe it would make it more clear. In this case changing the coordinate inside the class might make sense.

Reply:
Done